### PR TITLE
Enum - HeaderFlags, LoadFlags, SaveFlags, S5FixFlags

### PR DESCRIPTION
**Pull request by niceeffort**
Converting enum flags to new style per #1757

### CODE REVIEW

File: src/OpenLoco/src/S5/S5.h [hunk 1]
```diff
@@ -410,23 +424,36 @@ namespace OpenLoco::S5
...
+    enum class LoadFlags : uint32_t
+    {
+        none = 0U,
+        titleSequence = 1U << 0,
+        twoPlayer = 1U << 1,
+        scenario = 1U << 2,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(LoadFlags);
+
+    constexpr bool hasLoadFlags(LoadFlags flags, LoadFlags flagsToTest)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/S5/S5.h:436]: The load and save flags are just for the load and save functions they shouldn't need to be in the header.

Comment (duncanspumpkin) [reply src/OpenLoco/src/S5/S5.h:436]: Ahhh i remember now what my comment was about. Its the hasLoad/SaveFlags functions. Move them into the .cpp file they shouldn't be in the header as they aren't used by anything apart from the one .cpp file.

File: src/OpenLoco/src/S5/S5.cpp [hunk 2]
```diff
@@ -319,7 +337,7 @@ namespace OpenLoco::S5
                 fs.writeChunk(SawyerEncoding::runLengthSingle, file.gameState);
             }
 
-            if (file.header.flags & SaveFlags::raw)
+            if (static_cast<uint32_t>(file.header.flags) & static_cast<uint32_t>(SaveFlags::raw))
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/S5/S5.cpp:340]: this is actually a mistake. It should be `file.header.hasFlags(HeaderFlags::isRaw)`

**Review by duncanspumpkin — APPROVED** [review]
There is a minor mistake from upstream but thats it

**Comment by AaronVanGeffen** [comment]
@duncanspumpkin Could you rebase this and take care of the mistake?

**Comment by niceeffort** [comment]
> @duncanspumpkin Could you rebase this and take care of the mistake?

@duncanspumpkin Let me try the rebase. I think I can find some time this weekend and I would like to learn to do a rebase correctly.

**Comment by duncanspumpkin** [comment]
> > @duncanspumpkin Could you rebase this and take care of the mistake?
> 
> @duncanspumpkin Let me try the rebase. I think I can find some time this weekend and I would like to learn to do a rebase correctly.

already done it.

**Comment by niceeffort** [comment]
> > > @duncanspumpkin Could you rebase this and take care of the mistake?
> > 
> > 
> > @duncanspumpkin Let me try the rebase. I think I can find some time this weekend and I would like to learn to do a rebase correctly.
> 
> already done it.

Alright, I appreciate it.

File: src/OpenLoco/src/S5/S5.cpp [hunk 3]
```diff
@@ -226,26 +238,31 @@ namespace OpenLoco::S5
         return file;
     }
 
-    static constexpr bool shouldPackObjects(uint32_t flags)
+    static constexpr bool shouldPackObjects(SaveFlags flags)
     {
-        return !(flags & SaveFlags::raw) && !(flags & SaveFlags::dump) && (flags & SaveFlags::packCustomObjects) && !isNetworked();
+        return (flags & SaveFlags::raw) == SaveFlags::none
+            && (flags & SaveFlags::dump) == SaveFlags::none
+            && (flags & SaveFlags::packCustomObjects) != SaveFlags::none
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/S5/S5.cpp:245]: Should we use `hasSaveFlags` for these as well?

File: src/OpenLoco/src/S5/S5.cpp [hunk 4]
```diff
@@ -226,26 +238,31 @@ namespace OpenLoco::S5
...
         return exportGameStateToFile(fs, flags);
     }
 
-    bool exportGameStateToFile(Stream& stream, uint32_t flags)
+    bool exportGameStateToFile(Stream& stream, SaveFlags flags)
     {
-        if (!(flags & SaveFlags::noWindowClose) && !(flags & SaveFlags::raw) && !(flags & SaveFlags::dump))
+        if ((flags & SaveFlags::noWindowClose) == SaveFlags::none
+            && (flags & SaveFlags::raw) == SaveFlags::none
+            && (flags & SaveFlags::dump) == SaveFlags::none)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/S5/S5.cpp:260]: Should we use `hasSaveFlags` for these as well?

File: src/OpenLoco/src/S5/S5.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/S5/S5.cpp:245]: Ah, those functions are methods, so unavailable without making changes to what's passed in as a parameter. Let's leave it then.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**